### PR TITLE
Tracing analysis overview

### LINKED ISSUE #1337: open/active/done
Hi
something to handle open tasks in a project would be nice.
a typical workflow for a project is this:
- 1000 tasks are created and after one week 960 are finished.
- the last tasks always last very long - we need to contact the hiwis and ask to finish those tasks

So it would be really nice to have 
- the feature Kevin requested (#1335)
- in projects menu we have Open Assignments, here it would be great to have open/active/done instead (I can tell the scientist that his project is finished when open AND active is 0)
- ideally the number showing "active" is clickable and lists all active tasks to see who has how many and which tasks. So we would be able to contact them.

That would be really helpful to get complete projects done without  the scientist complaining that only 98% of his tasks are finished

Cheers
Heiko

 Reported by: Heiko Wissler (hwissler@brain.mpg.de) on master

<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/1337/create?referer=github" target="_blank">Log Time</a>

### LINKED ISSUE #2222: Create Hiwi Administration Overview pages
See https://discuss.webknossos.org/t/scripts-for-hiwi-administration/607

<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/2222/create?referer=github" target="_blank">Log Time</a>

### PR DESCRIPTION
**Pull request by fm3**
Add hiwi administration pages: Project Progress and Open Tasks.

**Needs Database Evolution, collecting OpenAssignments**

Note: makes the assumption that there is *one* `OpenAssignment` per Task in the database. @jfrohnhofen can we merge duplicate ones (summing their instances field)? That would also prepare us for #2230
Also related: #2198 

Note: uses new braingames-libs version 

### Steps to test:
- test new api routes `/api/teams/:id/openTasksOverview` and `/api/teams/:id/progressOverview`
- used by new pages under Statistics→Open Tasks and Statistics→Project Progress

### Issues:
- fixes #2222
- fixes #1337

------
- [x] Ready for review

### CODE REVIEW

**Comment by normanrz** [comment]
@hotzenklotz can you review the JS parts?

File: app/assets/javascripts/admin/api_flow_types.js [hunk 1]
```diff
@@ -236,4 +236,19 @@ export type APITimeTrackingType = {
   tasktype_summary: string,
 };
 
+export type APIProjectProgressReportType = Array<{
```
Comment (hotzenklotz) [inline app/assets/javascripts/admin/api_flow_types.js]: I would prefer type to be of a single instance. Then we have the flexibility to use them in arrays or single objects as need. Additionally, return types and variable declaration will inform you clearly that the are an array type.

**Comment by fm3** [comment]
conf/evolutions/36.js contains a database evolution to aggregate the OpenAssignments (so that then there is exactly one per task).
Commit 580c6cd modifies the code so that no new duplicate ones should be created

**Comment by fm3** [comment]
@jfrohnhofen could you review the scala parts?

File: app/controllers/ReportController.scala [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+case class ProjectProgressEntry(projectName: String, totalTasks: Int, totalInstances: Int, openInstances: Int,
+                                finishedInstances: Int, inProgressInstances: Int)
+object ProjectProgressEntry { implicit val jsonFormat = Json.format[ProjectProgressEntry] }
+
+class ReportController @Inject()(val messagesApi: MessagesApi) extends Controller with FoxImplicits {
+
+  def projectProgressOverview(teamId: String) = SecuredAction.async { implicit request =>
+    for {
+      team <- TeamDAO.findOneById(teamId)(GlobalAccessContext) ?~> "team.notFound"
+      t0 = System.nanoTime()
```
Comment (jfrohnhofen) [inline app/controllers/ReportController.scala]: Remove? this looks like a leftover from some measurement.

File: app/controllers/ReportController.scala [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      users <- UserDAO.findByTeams(List(Some(team.name), team.parent).flatten, false)
```
Comment (jfrohnhofen) [inline app/controllers/ReportController.scala]: Save the team list as a val, so we don't need to compute it twice

File: app/controllers/ReportController.scala [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+      projects <- ProjectDAO.findAllByTeamNames(List(Some(team.name), team.parent).flatten)(GlobalAccessContext)
+      entryBoxes <- Fox.sequence(projects.map(p => progressOfProject(p, users)(GlobalAccessContext)))
+    } yield {
+      Ok(Json.toJson(entryBoxes.flatten))
+    }
+  }
+
+  def progressOfProject(project: Project, users: List[User])(implicit ctx: DBAccessContext): Fox[ProjectProgressEntry] = {
```
Comment (jfrohnhofen) [inline app/controllers/ReportController.scala]: private (same for most of the other functions below)

File: app/controllers/ReportController.scala [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+  /**
+    * assumes that (a) there is only one OpenAssignment per Task
+    * and (b) that all tasks of a project have the same required experience
+    */
+  def openTasksOverview(id: String) = SecuredAction.async { implicit request =>
+    for {
+      team <- TeamDAO.findOneById(id)(GlobalAccessContext)
+      users <- UserDAO.findByTeams(List(team.name), true)(GlobalAccessContext)
+      nonAdminUsers = users.filter(!_.isAdminOf(team.name))
```
Comment (jfrohnhofen) [inline app/controllers/ReportController.scala]: `filterNot` (nitpick, I know)

**Comment by jfrohnhofen** [comment]
looks very good. feel free to merge @fm3

**Pull request merged**